fish-rust/src/fds.rs

```diff
@@ -1,13 +1,30 @@
 use crate::ffi;
 use nix::unistd;
-use std::os::unix::io::RawFd;
+use std::os::unix::io::{AsRawFd, FromRawFd, RawFd};
 
 /// A helper type for managing and automatically closing a file descriptor
-pub struct autoclose_fd_t {
+///
+/// This was implemented in rust as a port of the existing C++ code but it didn't take its place
+/// (yet) and there's still the original cpp implementation in `src/fds.h`, so its name is
+/// disambiguated because some code uses a mix of both for interop purposes.
+pub struct AutoCloseFd2 {
     fd_: RawFd,
 }
 
-impl autoclose_fd_t {
+#[cxx::bridge]
+mod autoclose_fd_t {
+    extern "Rust" {
+        #[cxx_name = "autoclose_fd_t2"]
+        type AutoCloseFd2;
```

Comment:
I'd much rather call this `AutoCloseFd` (or `AutocloseFd`?) and use this name also on the C++ side.
Then it's obvious where it's from.

Reply:
Not a bad idea.  Then we don’t need a second rename.